src/_optPlug.node/__xmlValidators/libxml2-wasm.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+This file is part of CycloneDX JavaScript Library.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+Copyright (c) OWASP Foundation. All Rights Reserved.
+*/
+
+import { readFile } from 'fs/promises';
+import { ParseOption, XmlDocument, XsdValidator } from 'libxml2-wasm';
+import { pathToFileURL } from 'url';
+
+import type { ValidationError } from '../../validation/types';
+import type { Functionality, Validator } from '../xmlValidator';
+
+/** @internal */
+export default (async function (schemaPath: string): Promise<Validator> {
+  const schema = XmlDocument.fromString(
+    await readFile(schemaPath, 'utf-8'),
+    {
+      option: ParseOption.XML_PARSE_NONET | ParseOption.XML_PARSE_COMPACT,
+      url: pathToFileURL(schemaPath).toString()
+    });
+  const validator = XsdValidator.fromDoc(schema);
+
+  return function (data: string): null | ValidationError {
+    const doc = XmlDocument.fromString(data, { option: ParseOption.XML_PARSE_NONET | ParseOption.XML_PARSE_COMPACT });
+    let errors = null;
+    try {
+      validator.validate(doc);
+    }
+    catch (validationErrors) {
+      errors = validationErrors;
+    }
+
+    doc.dispose();
+    validator.dispose();
```

Comment:
really free/dispose the validator and schema here?

Reply:
Good point, this will go wrong the second call. I could just not dispose the validator and the schema. But the library emphasises proper disposing (https://jameslan.github.io/libxml2-wasm/v0.4/documents/Memory_Management.html). Here I'm really lacking in Typescript knowledge on how to solve this,

Reply:
regarding `using`, read here: https://www.totaltypescript.com/typescript-5-2-new-keyword-using 

regarding manually disposing/freeing: maybe just try it out. in the end, it all is javascript - just see what you can do.

Reply:
@SierraNL , could you see if changing this makes the tests pass?